src/notify_service.cpp

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#include "config.h"
+
+#include "notify_service.hpp"
+
+#include <nlohmann/json.hpp>
+#include <phosphor-logging/lg2.hpp>
+
+#include <fstream>
+#include <iostream>
+#include <map>
+
+namespace data_sync::notify
+{
+namespace file_operations
+{
+nlohmann::json readFromFile(const fs::path& notifyFilePath)
+{
+    std::ifstream notifyFile;
+    notifyFile.open(fs::path(NOTIFY_SERVICES_DIR) / notifyFilePath);
+    nlohmann::json notifyInfoJSON(nlohmann::json::parse(notifyFile));
+
+    return notifyInfoJSON;
+}
+} // namespace file_operations
+
+NotifyService::NotifyService(sdbusplus::async::context& ctx,
+                             const fs::path& notifyFilePath)
+{
+    ctx.spawn(init(ctx, notifyFilePath));
+}
+
+sdbusplus::async::task<>
+    // NOLINTNEXTLINE
+    NotifyService::sendSystemDNotification(
+        sdbusplus::async::context& ctx,
+        const std::vector<std::string>& services, const std::string& method)
+{
+    for (const auto& service : services)
+    {
+        auto systemdReload = sdbusplus::async::proxy()
+                                 .service("org.freedesktop.systemd1")
+                                 .path("/org/freedesktop/systemd1")
+                                 .interface("org.freedesktop.systemd1.Manager");
+        try
+        {
+            using objectPath = sdbusplus::message::object_path;
+            if (method == "Restart")
+            {
+                lg2::debug("Restart request send for {SERVICE}", "SERVICE",
+                           service);
+                co_await systemdReload.call<objectPath>(ctx, "RestartUnit",
+                                                        service, "replace");
+            }
+            else if (method == "Reload")
+            {
+                lg2::debug("Reload request send for {SERVICE}", "SERVICE",
+                           service);
+                co_await systemdReload.call<objectPath>(ctx, "ReloadUnit",
+                                                        service, "replace");
+            }
+        }
+        catch (const sdbusplus::exception::SdBusError& e)
+        {
+            lg2::error(
+                "Failed to send systemd notification request to {SERVICE}: {ERROR}",
```

Comment:
Good to have filepath details , I feel we could able to get the NofifyInfo data.